tests/playbooks/tests_802_1x.yml

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+  vars:
+    interface: 802-1x-test
+    type: veth
+  tasks:
+    - name: "INIT: 802_1x tests"
+      debug:
+        msg: "##################################################"
+    - include_tasks: tasks/show-interfaces.yml
+    - include_tasks: tasks/manage-test-interface.yml
+      vars:
+        state: present
+    - include_tasks: tasks/assert-device_present.yml
+    - block:
+        - name: "TEST: I can create a profile with 802.1x"
+          debug:
+            msg: "##################################################"
+        - import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface }}"
+                state: up
+                type: ethernet
+                ip:
+                  dhcp4: "no"
+                  auto6: "no"
+                # "wait: 0" is required as the connection will fail to activate.
+                # The test environment doesn't have an actual 802.1x EAP provider.
```

Comment:
The NetworkManager CI has a script to setup hostapd with veth to test this scenario.

https://gitlab.freedesktop.org/NetworkManager/NetworkManager-ci/-/blob/master/prepare/hostapd_wired.sh
https://gitlab.freedesktop.org/NetworkManager/NetworkManager-ci/-/blob/master/nmcli/features/ethernet.feature#L417

Could you use this information to setup hostapd before this playbook and clean it up in the cleanup block?

Reply:
Thanks for the review guys.

I had at getting the NetworkManager-ci test environment working without much success. My main issue was getting traffic to go between two bridged veth interfaces. I was trying this even without any 802.1x auth and the only way i could get it working was using network namespaces:

```
ip link add veth1 type veth peer name veth1-br
ip link add veth2 type veth peer name veth2-br

ip link add br1 type bridge
ip link set br1 up

ip netns add ns1
ip netns add ns2

ip link set veth1 netns ns1
ip link set veth2 netns ns2

ip netns exec ns1 ip addr add 10.1.1.11/24 dev veth1
ip netns exec ns2 ip addr add 10.1.1.12/24 dev veth2

ip link set veth1-br up
ip link set veth2-br up

ip link set veth1-br master br1
ip link set veth2-br master br1

ip netns exec ns1 ip link set veth1 up
ip netns exec ns2 ip link set veth2 up

ip netns exec ns1 ping -c3 10.1.1.12
ip netns exec ns2 ping -c3 10.1.1.11
```

Is there anything I should be doing different to get this working without namespaces?

Reply:
Where did you try this? Might there be a firewall causing problems?

Reply:
Main thing I noticed when not using namespaces is I couldn't get ARP to go over the bridge, it just hit the bridge and didn't go anywhere...

Reply:
Sorry, didn't see your prev message. Firewalld is stopped and I did an iptables -F, doesn't seem to make a difference.
Was trying this on a rhel7 vm

Reply:
My method that doesn't work is this:
```
ip link add veth1 type veth peer name veth1-br
ip link add veth2 type veth peer name veth2-br

ip link add br1 type bridge
ip link set br1 up


ip addr add 10.1.1.11/24 dev veth1
ip addr add 10.1.1.12/24 dev veth2

ip link set veth1-br up
ip link set veth2-br up

ip link set veth1-br master br1
ip link set veth2-br master br1

ip link set veth1 up
ip link set veth2 up
```

tcpdump on br1 looks like this when i run `ping 10.1.1.12 -I veth1`:
```
[root@rhel7-test ~]# tcpdump -enni br1
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on br1, link-type EN10MB (Ethernet), capture size 262144 bytes
08:03:54.949716 ea:8c:4f:b2:d9:9d > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.1.1.12 tell 10.1.1.11, length 28
08:03:55.951722 ea:8c:4f:b2:d9:9d > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.1.1.12 tell 10.1.1.11, length 28
08:03:57.948878 ea:8c:4f:b2:d9:9d > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.1.1.12 tell 10.1.1.11, length 28
08:03:58.949938 ea:8c:4f:b2:d9:9d > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.1.1.12 tell 10.1.1.11, length 28
08:03:59.951731 ea:8c:4f:b2:d9:9d > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.1.1.12 tell 10.1.1.11, length 28
```

Reply:
@vbenes do you have any idea?

Reply:
Talked to a guy at work today and he said we should be all good to just create one interface in a namespace and leave the other in the default namespace. Then we can run the hostapd process in the secondary namespace and the ansible role can configure 802.1x on the interface in the default namespace.

Managed to get this to work:
```
ip netns exec ns1 ip addr add 10.1.1.11/24 dev veth1
ip addr add 10.1.1.12/24 dev veth2

ip link set veth1-br up
ip link set veth2-br up

ip link set veth1-br master br1
ip link set veth2-br master br1


ip netns exec ns1 ip link set veth1 up
ip link set veth2 up

ping -c3 10.1.1.11 -I veth2
```

Next step will be to try to run hostapd like this:
```
ip netns exec ns1 hostapd --arg1 --arg2
````

Reply:
Managed to get hostapd to run in a namespace and could successfully connect to it with an 802.1x interface created using the role. Made a playbook to set everything up but it needs a little more work. Will aim to add another commit to the PR tomorrow. 

Does anyone know if the test vms have the EPEL repo enabled? Otherwise I will configure this as part of the playbook.

Reply:
EPEL is not enabled by default.